src/components/ImageTransformer/ImageTransformer.js

```diff
@@ -89,36 +89,70 @@ export default class TransformerComponent extends Component {
     this.transformer.getLayer().batchDraw();
   }
 
+  fitBBoxToScaledStage(box, stage) {
+    let { x, y, width, height } = box;
+
+    const [realX, realY] = [box.x + stage.x, box.y + stage.y];
+
+    if (realX < 0) {
+      x = 0;
+      width += realX;
+    } else if (realX + box.width > stage.width) {
+      width = stage.width - realX;
+    }
+
+    if (realY < 0) {
+      y = 0;
+      height += realY;
+    } else if (realY + box.height > stage.height) {
+      height = stage.height - realY;
+    }
+
+    return { ...box, x, y, width, height };
+  }
+
+  getStageAbsoluteDimensions() {
+    const stage = this.transformer.getStage();
+    const [scaledStageWidth, scaledStageHeight] = [stage.width() * stage.scaleX(), stage.height() * stage.scaleY()];
+    const [stageX, stageY] = [stage.x(), stage.y()].map(Math.abs);
```

Comment:
would be more correct to leave them as is and subtract them in `fitBBoxToScaledStage()` because this math is based on assumption that we'll never move stage to positive coordinates, but we can change this later.

Reply:
fair point